docs/build-nix.md

```diff
@@ -0,0 +1,66 @@
+# Building on Nix
+
+The Nix Ecosystem insists on the purity of its builds, therefore compiling 
+Dosbox Staging on it requires slightly different inputs so meson gets the 
+appropriate environment variables from pkg-config.
```

Comment:
> insists on the purity of its builds

I would suggest rephrasing this; it sounds more like a jab (I know that's not the intent).

Either explain why nixos is extra hard to build with (their goal that builds must be hash-level reproduce means that all dependencies need to have exact versions specified - as well as dependencies of dependencies, and so on). 

Another option is just linking to their discussion about builds.